python3/tests/test_dnf_plugins.py

```diff
@@ -0,0 +1,96 @@
+"""Test module for dnf accesstoken"""
+import unittest
+import sys
+import json
+from unittest.mock import MagicMock, patch
+
+sys.modules["urlgrabber"] = MagicMock()
+
+# Disable wrong import postition as need to mock some sys modules first
+#pylint: disable=wrong-import-position
+
+# Disable unused-argument as some mock obj is not used
+#pylint: disable=unused-argument
+
+# Some test case does not use self
+#pylint: disable=no-self-use
+
+from dnf_plugins import accesstoken
+from dnf_plugins import ptoken
+
+REPO_NAME = "testrepo"
+
+def _mock_repo(a_token=None, p_token=None, baseurl=None):
+    mock_repo = MagicMock()
+    mock_repo.accesstoken = a_token
+    mock_repo.ptoken = p_token
+    mock_repo.baseurl = baseurl
+    mock_base = MagicMock()
+    mock_base.repos = {REPO_NAME: mock_repo}
+    mock_repo.base = mock_base
+    return mock_repo
+
+
+@patch("dnf_plugins.accesstoken.urlgrabber")
+class TestAccesstoken(unittest.TestCase):
+    """Test class for dnf access plugin"""
+
+    def test_set_http_header_with_access_token(self, mock_grabber):
+        """test config succeed with accesstokan"""
+        mock_repo = _mock_repo(a_token="file:///mock_accesstoken_url")
+        mock_grabber.urlopen.return_value.read.return_value = json.dumps({
+            "token": "valid_token",
+            "token_id": "valid_token_id",
+        })
+        accesstoken.AccessToken(mock_repo.base, MagicMock()).config()
+        mock_repo.set_http_headers.assert_called_with(
+            ['X-Access-Token:valid_token','Referer:valid_token_id']
+        )
+
+    def test_repo_without_access_token(self, mock_grabber):
+        """If repo has not accestoken, it should not be blocked"""
+        mock_repo = _mock_repo()
+        accesstoken.AccessToken(mock_repo.base, MagicMock()).config()
```

Comment:
Need a `assert_not_called `?

Reply:
As comments, the target is to test repo without accesstoken still work, so this case does not care whether it is called or not,
but it has no bad to put a no called there anyway.

Reply:
Edit: just updated to follow the suggestion.